### PR TITLE
Migrating CircleCI Jobs to GHA Workflow

### PR DESCRIPTION
**Pull request by marcboudreau**
This PR introduces a **diff-oss-ci** and a **test-ui** GitHub Actions Workflow Job based on the CircleCI jobs bearing the same names.

The **diff-oss-ci** job is skipped when the workflow executions stems from a repository other than `hashicorp/vault-enterprise`.

This PR also introduces a few adjustments to satisfy issues found by the GitHub Actions linter.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -101,3 +135,96 @@ jobs:
...
+      name: build-go-dev
+      run: |
+        rm -rf ./pkg
+        mkdir ./pkg
+
+        make ci-bootstrap dev
+    - id: test-ui
+      name: test-ui
+      env:
+        VAULT_LICENSE: ${{ steps.secrets.outputs.VAULT_LICENSE }}
```
Comment (miagilepner) [inline .github/workflows/ci.yml:209]: what happens when the `secrets` step is skipped?

File: .github/workflows/actionlint.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  push:
+  pull_request:
+
+jobs:
+  actionlint:
+    runs-on: ['self-hosted', 'linux', 'small']
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Check workflow files"
+        uses: docker://docker.mirror.hashicorp.services/rhysd/actionlint:latest
```
Comment (miagilepner) [inline .github/workflows/actionlint.yml:13]: should we specify a version here, rather than using latest?

File: .github/workflows/actionlint.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+name: Lint GitHub Actions Workflows
+on:
+  push:
+  pull_request:
+
+jobs:
+  actionlint:
+    runs-on: ['self-hosted', 'linux', 'small']
+    steps:
+      - uses: actions/checkout@v3
```
Comment (miagilepner) [inline .github/workflows/actionlint.yml:11]: other workflows use a commit hash for the version. should we do the same here?

File: .github/workflows/actionlint.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+
+name: Lint GitHub Actions Workflows
+on:
+  push:
+  pull_request:
```
Comment (stevendpclark) [inline .github/workflows/actionlint.yml:5]: Possibly consider limiting runs of this workflow only to when things it cares about change?

```
    paths:
      - .github/**
```

File: .github/workflows/test-run-acc-tests-for-path.yml [hunk 5]
```diff
@@ -22,7 +22,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - id: get-metadata
-        run: echo "go-version=$(cat ./.go-version)" >> $GITHUB_OUTPUT
+        run: echo "go-version=$(cat ./.go-version)" >> "$GITHUB_OUTPUT"
       - name: Set Up Go
         uses: actions/setup-go@v3
         with:
```
Comment (kubawi) [inline .github/workflows/test-run-acc-tests-for-path.yml:28]: :thinking: We should be able to get rid of the `get-metadata` step here, see comment on `.github/workflows/godoc-test-checker.yml` 😄

File: .github/workflows/actionlint.yml (hunk 2 above)
Comment (marcboudreau) [reply .github/workflows/actionlint.yml]: Yes, definitely. I've fixed it.

File: .github/workflows/godoc-test-checker.yml [hunk 6]
```diff
@@ -17,11 +17,11 @@ jobs:
           fetch-depth: 0
       - name: get metadata
         id: get-metadata
-        run: echo "go-version=$(cat ./.go-version)" >> $GITHUB_OUTPUT
+        run: echo "go-version=$(cat ./.go-version)" >> "$GITHUB_OUTPUT"
       - name: Set Up Go
         uses: actions/setup-go@v3
         with:
           cache: true
           go-version: ${{ steps.get-metadata.outputs.go-version }}
```
[the remark this answers is not in the document]
Comment (marcboudreau) [reply .github/workflows/godoc-test-checker.yml]: I've fixed this and removed the get-metadata step as a result.

File: .github/workflows/godoc-test-checker.yml [hunk 7]
```diff
@@ -17,11 +17,11 @@ jobs:
           fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (marcboudreau) [reply .github/workflows/godoc-test-checker.yml:16]: Ok, thanks. I'll remove it then.

**Pull request merged**